Updates to clarify usage of COA signing keys (#1155)
onflow/docs: docs/networks/node-ops/evm-gateway/evm-gateway-setup.md

### Account and Key Management

When operating an EVM Gateway it is important to understand how keys are configured and used. Each gateway instance 

> must be configured with a Flow account address, which is funded and which it uses to pay to wrap EVM transactions into a 

Reviewer: ```suggestion
must be configured with a Flow account address, which is sufficiently funded and which it uses to pay to wrap EVM transactions into a 
```

Cadence transaction when submitting to the Flow Access Node. This can be configured with a standalone private key file for 
the `--coa-key` config flag. Alternatively, you may also use cloud KMS providers per the guidance below.

#### COA Account signing key rotation

The gateway implements a signing key rotation scheme to scale the use of the `COA_KEY` and enable it to be used across many
EVM client transactions. This is configured by reusing the COA public key to create new on-chain signing keys. Although it may seem
counter-intuitive to use the same COA public key when creating new signing keys, the newly added keys all occupy different 
key slots on the account which enables the gateway to support concurrent transaction signing without causing nonce collisions 

> for EVM clients. 

Author: Yeah sure, where would you want to link it?

> The following transaction creates 100 keys on the account signing the transaction to add these keys:

Reviewer 2: we should add something like the following

> Assuming there is already one key on the account, this transaction adds 100 copies of the first key

```swift
transaction {
    prepare(signer: auth(AddKey) &Account) {
		let firstKey = signer.keys.get(keyIndex: 0)!

> 		let range: InclusiveRange<Int> = InclusiveRange(1, [100], step: 1)

Author: ```suggestion
		let range: InclusiveRange<Int> = InclusiveRange(1, 100, step: 1)
```

		for element in range {
			signer.keys.add(
				publicKey: firstKey.publicKey,
				hashAlgorithm: HashAlgorithm.SHA2_256,
				weight: 1000.0
			)
		}
	}
}
```

> Signing keys which are added to the COA account are required to use the `SHA2_256` hashing algorithm.

Reviewer 2: it doesn't have to be `SHA2_256`, it just has to match the hashing algorithm of the key that we're copying. For us at the foundation, it always has to be sha2 256, but that's not universally true

Author: Ah ok. Ardit had said that previously. Will update

Reviewer 3: Cloud KMS Keys support only the `SHA2_256` hashing algorithm, that's why we used this one.

:::note

If you are operating your EVM Gateway(s) to relay traffic for Flow EVM, or if you otherwise anticipate high volumes of 
transactions we recommend configuring at least 200 signing keys or more. Signing key utilization increases proportionately

> with grown the transaction throughput. You can track signing key utilization as a metric, see below. 

Author: ```suggestion
with transaction throughput growth. You can track signing key utilization as a metric, see below. 
```

		let firstKey = signer.keys.get(keyIndex: 0)!
		let range: InclusiveRange<Int> = InclusiveRange(1, 100, step: 1)
		for element in range {
			signer.keys.add(
				publicKey: firstKey.publicKey,
				hashAlgorithm: HashAlgorithm.SHA2_256,
				weight: 1000.0
			)
		}
	}
}
```
Signing keys which are added to the COA account are required to use the same hashing algorithm as the key being copied.

If keys are added while the gateway is running it will need to be restarted to make use of the new keys.

:::note

If you are operating your EVM Gateway(s) to relay traffic for Flow EVM, or if you otherwise anticipate high volumes of 
transactions we recommend configuring at least 200 signing keys or more. Signing key utilization increases proportionately
with transaction throughput growth. You can track signing key utilization as a metric, see `evm_gateway_available_signing_keys` below. 

:::

#### KMS Configuration

EVM Gateway allows for Google and AWS Key Management Service (KMS) integration, which is the recommended way of setting up the gateway 
for live networks. It is only required to configure a single KMS key for the Flow account configured as the gateway `COA_ACCOUNT`. 

> This key is not the same as those used by the gateway for transaction signing, as outlined above. It is only used to fund 
> transactions passed to the Access Node.

Author: ```suggestion
```
